### ISSUE TITLE
Clean up p2 update storage

### ISSUE THREAD

**Issue by kthoms** [eclipse/xtext#1626]
https://download.eclipse.org/modeling/tmf/xtext/updates/
contains several composites, some of them seem obsolete.

This report shows that one of them actually broke:
https://download.eclipse.org/oomph/archive/p2-index/modeling.tmf.xtext.html#err0

ATM there is no meaningful "maintenance" or "staging". Shall these locations be removed, archived, fixed?

This issue is to describe how the updates storage are cleaned up.

**Comment by cdietrich** [comment]
we should also consider to move some stuff to archive to relieve mirrors

**Comment by kthoms** [comment]
see also https://download.eclipse.org/oomph/archive/eclipse/modeling/tmf/xtext/

**Comment by kthoms** [comment]
We will delete the following composites (no need to archive them):
- http://download.eclipse.org/modeling/tmf/xtext/updates/composite/staging
- http://download.eclipse.org/modeling/tmf/xtext/updates/composite/maintenance
- http://download.eclipse.org/modeling/tmf/xtext/updates/composite/epp

**Comment by cdietrich** [comment]
any new @Hejado @kthoms ?

**Comment by cdietrich** [comment]
move to 2.22

**Comment by Hejado** [comment]
I deleted these two:

* http://download.eclipse.org/modeling/tmf/xtext/updates/composite/staging
* http://download.eclipse.org/modeling/tmf/xtext/updates/composite/maintenance

However, http://download.eclipse.org/modeling/tmf/xtext/updates/composite/epp does not exist. https://download.eclipse.org/modeling/tmf/xtext/updates/epp/ does. Should I delete this one as well?

Just want to make sure.

**Comment by cdietrich** [comment]
@Hejado https://download.eclipse.org/modeling/tmf/xtext/updates/epp/ does exist.

**Comment by cdietrich** [comment]
@kthoms are we sure noone uses the epp thing? what does it contain?

**Comment by Hejado** [comment]
Yes, it does. https://download.eclipse.org/modeling/tmf/xtext/updates/**composite**/epp does not.

**Comment by Hejado** [comment]
```<children size="3">
<child location="../releases/2.9.2/"/>
<child location="http://download.eclipse.org/releases/mars/201602261000/"/>
<child location="http://download.itemis.com/updates/releases/2.1.1/"/>
</children>```

**Comment by cdietrich** [comment]
hmmm wonder if anybody uses that old crap ...

**Comment by kthoms** [comment]
Looks outdated.

**Comment by cdietrich** [comment]
can we backup the stuff somewhere?

**Comment by kthoms** [comment]
> hmmm wonder if anybody uses that old crap ...

If so: Punish them 😈

**Comment by kthoms** [comment]
> can we backup the stuff somewhere?

Could move to archive, but I doubt anyone will use or need it there.

**Comment by cdietrich** [comment]
a local backup would be sufficient i think

**Comment by Hejado** [comment]
it's a text file, nothing more

**Comment by Hejado** [comment]
the referenced release itself (https://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.9.2/) can be kept/archived.

**Comment by Hejado** [comment]
my machine is not sufficient for a backup, because nobody but me has access.

**Comment by cdietrich** [comment]
then please attach the files here

**Comment by Hejado** [comment]
Taken from: https://download.eclipse.org/modeling/tmf/xtext/updates/epp/

[compositeArtifacts.xml.txt](https://github.com/eclipse/xtext/files/4275124/compositeArtifacts.xml.txt)
[compositeContent.xml.txt](https://github.com/eclipse/xtext/files/4275125/compositeContent.xml.txt)

(Had to rename, because GitHub does not support xml, only txt. :smirk:)

**Comment by Hejado** [comment]
Removed updates/epp here: https://ci.eclipse.org/xtext/job/releng/job/manual-one-off-commands/6/console

What about updates/archive/? Is this necessary instead of using archives.eclipse.org?

Also, updates/composite/* has sites that sound like the ones I deleted.

**Comment by cdietrich** [comment]
i dont think we need composites for archives. people using old stuff should pick a specific repo

**Comment by cdietrich** [comment]
as we also seem to create broken composite we should have 1-3 jobs validating them by calling tycho on a respective target

**Comment by cdietrich** [comment]
move to 2.24

**Comment by cdietrich** [comment]
move to 2.25

**Comment by cdietrich** [comment]
move to 2.26

**Comment by NicoPrediger** [comment]
Unassigning myself due to priorities. Re-assign if priorities change...
